app/celery/scheduled_tasks.py

```diff
@@ -206,15 +207,29 @@ def _get_dynamodb_comp_pen_messages(
     message_limit: int,
 ) -> list:
     """
-    Helper function to get the Comp and Pen data from our dynamodb cache table.
+    Helper function to get the Comp and Pen data from our dynamodb cache table.  Items should be returned if all of
+    these attribute conditions are met:
+        1) is_processed is not set or False
+        2) has_duplicate_mappings is not set or False
+        3) payment_id is not equal to -1 (placeholder value)
+        4) paymentAmount exists
 
     :param table: the dynamodb table to grab the data from
     :param message_limit: the number of rows to search at a time and the max number of items that should be returned
     :return: a list of entries from the table that have not been processed yet
+
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/dynamodb.html#querying-and-scanning
     """
 
-    results = table.scan(FilterExpression=boto3.dynamodb.conditions.Attr('is_processed').eq(False), Limit=message_limit)
+    filters = (
+        Attr('is_processed').eq(False)
+        & Attr('payment_id').exists()
+        & Attr('payment_id').ne(-1)
```

Comment:
Should the value of `payment_id` be a constant? I saw we used a constant in the Consumer repo.